server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/model/MailboxFactory.scala

```diff
@@ -152,7 +163,60 @@ class MailboxFactory @Inject() (subscriptionManager: SubscriptionManager) {
             namespace = namespace,
             rights = rights,
             quotas = quotas,
-            isSubscribed = retrieveIsSubscribed)))
+            isSubscribed = isSubscribed))
+    }
+  }
+
+  def create(id: MailboxId, mailboxSession: MailboxSession, quotaLoader: QuotaLoader): SMono[Mailbox] = {
+    try {
+      val messageManager: MessageManager = mailboxManager.getMailbox(id, mailboxSession)
+      val resolvedACL = messageManager.getResolvedAcl(mailboxSession)
+
+      val name: Either[IllegalArgumentException, MailboxName] = retrieveMailboxName(messageManager.getMailboxPath, mailboxSession)
+
+      val role: Option[Role] = getRole(messageManager.getMailboxPath, mailboxSession)
+      val sortOrder: SortOrder = getSortOrder(role)
+      val quotas: SMono[Quotas] = quotaLoader.getQuotas(messageManager.getMailboxPath)
+      val rights: Rights = getRights(resolvedACL)
+
+      val sanitizedCounters: MailboxCounters = messageManager.getMailboxCounters(mailboxSession).sanitize()
+      val unreadEmails: Either[NumberFormatException, UnsignedInt] = UnsignedInt.validate(sanitizedCounters.getUnseen)
```

Comment:
it's a waste to compute all the Either, you should short-circuit the method in case any is Left.
For that, you should use a for comprehension here

Reply:
Should use a what? Missing a word here?

Reply:
https://docs.scala-lang.org/tour/for-comprehensions.html

Reply:
Sure ok. It makes the `MailboxValidation` useless then?

Reply:
probably

Reply:
I don't see why using for comprehension is useful in this case then? Or I might not understand correctly the for comprehensions then...

Let's say, on this line for example, if I do:

```
val unreadEmails: Either[NumberFormatException, UnreadEmails] = 
      for (unreadEmails <- UnsignedInt.validate(sanitizedCounters.getUnseen).map(UnreadEmails)) yield unreadEmails
```
I still have an Either right? On which if I want to short-circuit I probably to follow it by a match like:

```
val validatedUnreadEmails = unreadEmails match {
      case Left(error) => SMono.raiseError(error)
      case scala.Right(unreadEmails) => unreadEmails
}
```

Why not do a match directly? and I don't know in case of error I want to raise a Mono error, but in case not I just want an `UnreadEmails`, which are completely different types? I'm sorry I don't understand how you can achieve your suggestion and what's the interest of for comprehensions at this point

Reply:
Ok figured it out with @remk